src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java

```diff
@@ -124,21 +129,95 @@ public MavenMojoProjectParser(Log logger, Path baseDir, boolean pomCacheEnabled,
                         )))
                 .filter(Objects::nonNull)
                 .collect(toList());
-    }
 
-    @Nullable
-    private GitProvenance gitProvenance(Path baseDir, @Nullable BuildEnvironment buildEnvironment) {
-        try {
-            return GitProvenance.fromProjectDirectory(baseDir, buildEnvironment);
-        } catch (Exception e) {
-            // Logging at a low level as this is unlikely to happen except in non-git projects, where it is expected
-            logger.debug("Unable to determine git provenance", e);
+        JavaParser javaParser = JavaParser.fromJavaVersion()
+                .styles(styles)
+                .logCompilationWarningsAndErrors(false)
+                .build();
+
+        // Some annotation processors output generated sources to the /target directory
+        List<Path> generatedSourcePaths = listJavaSources(mavenProject.getBuild().getDirectory());
```

Comment:
If we parse generated sources as ASTs, we run the risk of generating diffs that don't accomplish anything when making a git commit.

Reply:
oh, great catch.